tests/unit/amo/sagas/test_collections.js

```diff
@@ -327,164 +328,248 @@ describe(__filename, () => {
     });
   });
 
-  describe('updateCollection', () => {
+  describe('modifyCollection', () => {
+    const formOverlayId = 'some-form-overlay';
+
+    const _createCollection = (params = {}) => {
+      sagaTester.dispatch(createCollection({
+        errorHandlerId: errorHandler.id,
+        formOverlayId,
+        name: 'some-collection-name',
+        slug,
+        user,
+        ...params,
+      }));
+    };
+
     const _updateCollection = (params = {}) => {
       sagaTester.dispatch(updateCollection({
         errorHandlerId: errorHandler.id,
-        formOverlayId: 'some-form-overlay',
+        formOverlayId,
         collectionSlug: 'some-collection',
-        user: 'some-user',
+        user,
         ...params,
       }));
     };
 
-    it('sends a patch to the collection API', async () => {
-      const params = {
-        collectionSlug: 'a-collection',
-        description: { 'en-US': 'New collection description' },
-        formOverlayId: 'some-form-overlay',
-        name: { 'en-US': 'New collection name' },
-        slug: 'new-slug',
-        user: 543,
-      };
-      const state = sagaTester.getState();
+    describe('common logic', () => {
+      // The common logic can be tested either by dispatching updateCollection
+      // or createCollection, so we just use _updateCollection.
+      it('clears the error handler', async () => {
+        _updateCollection();
 
-      mockApi
-        .expects('updateCollection')
-        .withArgs({
-          api: state.api,
-          collectionSlug: params.collectionSlug,
-          defaultLocale: undefined,
-          description: params.description,
-          name: params.name,
-          slug: params.slug,
-          user: params.user,
-        })
-        .once()
-        .returns(Promise.resolve());
+        const expectedAction = errorHandler.createClearingAction();
 
-      _updateCollection(params);
+        const action = await sagaTester.waitFor(expectedAction.type);
+        expect(action).toEqual(expectedAction);
+      });
 
-      const expectedAction = finishFormOverlaySubmit(params.formOverlayId);
+      it('begins and finishes a form submit', async () => {
+        mockApi.expects('updateCollection').returns(Promise.resolve());
+        _updateCollection({ formOverlayId });
```

Comment:
Hmm, yeah, I guess `formOverlayId` is used enough to warrant relying on the shared value. You might as well reap all the benefits and just leave it off everywhere:
```js
_updateCollection()
```